content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2385 @@
+  varbit-xor for all further samples, XOR'ing corresponding buckets from the
+  current and the previous sample.
+
+The following events trigger cutting a new chunk (for the reasons described in
+parentheses):
+
+- A change of sample type between integer histogram and float histogram
+  (because both require different chunk encodings altogether).
+- A change of sample type between gauge histogram and counter histogram
+  (because the leading byte has to denote the different type).
+- A counter reset for a counter histogram (to be stored in the leading byte as
+  counter reset information, see details below).
+- A schema change (which means we need a new chunk layout, and a chunk can only
+  have one chunk layout).
+- A change of the zero threshold (which changes the chunk layout, see above).
+- A change of the custom values (which changes the chunk layout, see above).
+- A staleness marker is followed by a regular sample (which does not strictly
+  require a new chunk, but it can be assumed that most histograms will change
+  so much when they go away and come back that cutting a new chunk is the best
+  option).
+- The chunk size limit is exceeded (see [details below](#chunk-size-limit)).
+
+Differences in the spans would also change the chunk layout, but they are
+reconciled by adding (explicitly represented) unpopulated buckets as needed so
+that all histograms in a chunk share the same span structure. This is
+straightforward if a bucket disappears, because the missing bucket is simply
+added to the new histogram as an unpopulated bucket while the histogram is
+appended to the chunk. However, disappearance of a formerly populated bucket
+constitutes a counter reset (see [below](#counter-reset-considerations)), so
+this case can only happen for gauge histograms (which do not feature counter
+resets). The far more common case is that buckets exist in a newly appended
+histogram that did not exist in the previously appended histograms. In this
+case, these buckets have to be added as explicitly unpopulated buckets to all
+previously appended histograms. This requires a complete re-encoding of the
+entire chunk. (There is some optimization potential in only re-encoding the
+affected parts. Implementing this would be quite complicated. So far, the
+performance impact of the full re-encoding did not stick out as problematic.)
+
+### Staleness markers
+
+Staleness markers used to be simply a special float value among the many that
+can be used to represent the `NaN` value. The same float value could be used to
+mark a histogram time series as stale. However, this would require cutting a
+new chunk, just for the purpose of marking the series as stale, because a float
+value following a histogram value has to be stored in a different chunk (see
+above). Therefore, there is also a histogram version of a stale marker where
+the field for the sum of observations is set to the special stale `NaN` value.
```

Comment:
```suggestion
the field for the sum of observations is set to the special stale `NaN` value. Note: this special `NaN` value is a float value different to other representations of `NaN`, in particular other `NaN` values that would arise from arithmetic operations, thus it is different from observing a "natural" `NaN` value as described in [Special cases of observed values](#special-cases-of-observed-values).
```

Reply:
I have tried to explain this in the first sentence of this section (albeit in a more condensed form). The confusion encountered here proves that we need a better explanation. However, I think it should be at the beginning of the section, not in the middle. See what I have done (in the next commit).

Reply:
thank you